src/main/scala/ignition/core/jobs/utils/SparkContextUtils.scala

```diff
@@ -194,5 +199,104 @@ object SparkContextUtils {
       else
         objectHadoopFile(paths, minimumPaths)
     }
+
+    def parallelTextFiles(paths: Seq[String], maxBytesPerPartition: Long): RDD[String] = {
+      val foundFiles = parallelListFiles(paths)
+      val files = foundFiles.map(_.path)
+
+      val s3Key = sc.hadoopConfiguration.get("fs.s3n.awsAccessKeyId")
+      val s3Secret = sc.hadoopConfiguration.get("fs.s3n.awsSecretAccessKey")
+
+      val partitionedFiles = sc.parallelize(files).map(file => file -> ()).partitionBy(createPartitioner(foundFiles, maxBytesPerPartition))
+
+      partitionedFiles.mapPartitions { files =>
+        val conf = new Configuration()
+        conf.set("fs.s3n.awsAccessKeyId", s3Key)
+        conf.set("fs.s3n.awsSecretAccessKey", s3Secret)
+        val codecFactory = new CompressionCodecFactory(conf)
+        files.map { case (path, _) => path } flatMap { s3Path =>
+          val fileSystem = FileSystem.get(new java.net.URI(s3Path), conf)
+          val path = new Path(s3Path)
+          val inputStream = Option(codecFactory.getCodec(path)) match {
+            case Some(compression) => compression.createInputStream(fileSystem.open(path))
+            case None => fileSystem.open(path)
+          }
+          try {
+            Source.fromInputStream(inputStream).getLines().toList
```

Comment:
LGTM

Personally I should complete replace any reference of io.Source apache IOUtils from anything serious until find a decent doc that said otherwise. 

Good words:  

"Scala's io library was just hack done to provide support for limited needs.... ...If you are going to use I/O for anything more than reading the occasional file on short-lived processes, you'd better either use Java libraries"
- http://stackoverflow.com/questions/4458864/whats-the-right-way-to-use-scala-io-source